### PR TITLE
[#177] Add support for building on Ubuntu 22.04 (main)

### PR DESCRIPTION
**Pull request by alanking**
\- Remove autoconf external as it is no longer used.

\- Bump required ruby version to 3.1.2 because there is no package for
the previous version available for Ubuntu 22.04 so we need to bump this
version in order to build externals for this platform.

\- Bump SHAs for the following externals in order to prevent build errors
caused by deprecation warnings due to using old OpenSSL types and
functionality:

  aws-sdk-cpp
  libs3
  qpid-proton

Ubuntu 22.04 uses OpenSSL 3.0 and the build processes for these
externals are compatible with 1.1.0.

---

Still trying on other platforms and messing with minimum bumps and whatnot, but this worked for building externals which can build iRODS packages which run in the testing environment.

Sibling PRs:
https://github.com/irods/aws-sdk-cpp/pull/1
https://github.com/irods/libs3/pull/22
https://github.com/irods/qpid-proton/pull/2

### CODE REVIEW

**Comment by trel** [comment]
autoconf was confirmed not used/needed for Centos7, our oldest current supported distribution?

**Comment by alanking** [comment]
At this point I have only confirmed for ubuntu 22.04 that any of this works, so I'll report back once I've watched this work on the other platforms. So far so good on centos:7 - installing prerequisites was fine and the externals are building now.

**Comment by SwooshyCueb** [comment]
If https://github.com/apache/qpid-proton/commit/5aaa655cb1023e9fb48e59f64e3d0c1307ea54f2 is really all that is needed, I'd rather just disable the warning/error with a compiler flag.
If https://github.com/apache/qpid-proton/commit/5aaa655cb1023e9fb48e59f64e3d0c1307ea54f2 *and all the commits leading up to it* are required, however, I'm fine with targeting this specific sha, but I'd rather keep our package version numbers sequential. I would propose we follow the pattern of `[last released version from the branch containing the commit]-[number of commits since that release]`. So something like `0.37.0-156`. (Normally a dash would be problematic, but since this version number is not actually used in the package metadata, just in the package name and a directory name.

**Comment by alanking** [comment]
Oh, I'm also playing around with disabling the compiler warning as is being done in that commit.

**Comment by alanking** [comment]
These changes successfully built externals for Ubuntu 22 (newest 4.3.x platform) and Centos 7 (oldest 4.3.x platform). Going to try building iRODS packages with them next.

Leaving this in draft for now as we work through issues, but I'm happy to put this in after some squashing/issue linking if desired.

**Comment by alanking** [comment]
iRODS packages built successfully for Ubuntu 22.04 and Centos 7 with these externals + changes in https://github.com/irods/irods/pull/6620 + changes in https://github.com/irods/irods_development_environment/pull/72. I confirmed that the packages install and run with https://github.com/irods/irods_testing_environment/pull/118.

So, pending https://github.com/irods/libs3/pull/22 being merged and appropriate SHA being introduced to `versions.json` here, I think this is ready. Marking as Ready for Review.

**Comment by alanking** [comment]
Oh, I plan on squashing to a single commit and linking to #177. Won't do that til I can fix the versions.json, as mentioned before.

File: versions.json
Comment (korydraughn) [inline versions.json:266]: Is this what we want for the commitish and version string?

I'm guessing this is a placeholder for the real thing because it points to your fork of libs3?

**Review by korydraughn — COMMENTED** [review]
The ruby version change certainly requires that we verify these changes work for all supported OS.

Just saying out loud in case no one has done that yet.

File: versions.json
Comment (alanking) [reply versions.json:266]: Yep, once we merge https://github.com/irods/libs3/pull/22, we can update the URL and the sha.

Comment (alanking) [reply versions.json]: Done

**Comment by alanking** [comment]
Okay, the only remaining concern I have here is the bumping of the AWS SDK.

I pointed to the official upstream and went to a more recent version that addresses the OpenSSL 3 issue. Like qpid-proton, we could just disable the warning instead of changing the SHA. I defer to @SwooshyCueb on the best path forward here. I'm okay with doing whatever seems best.

**Comment by SwooshyCueb** [comment]
Unless there's a reason not to, I'd say bump the sha

**Comment by alanking** [comment]
Sounds good. In that case, I think this is ready to go. Any other thoughts, @korydraughn or @trel?

**Comment by korydraughn** [comment]
How do you feel about updating our fork of AWS and pointing versions.json at that?

Is there a reason we're choosing to point at upstream rather than our own fork?

**Comment by alanking** [comment]
I pulled the 1.9.323 tag from aws/aws-sdk-cpp... and then pushed the 1.9.323 tag to irods/aws-sdk-cpp... and then confirmed that the SHAs match.

I am building aws-sdk-cpp once more just in case.

**Comment by alanking** [comment]
Building succeeded.

**Review by korydraughn — COMMENTED** [review]
Please squash to taste and leave off the pounds.

**Comment by alanking** [comment]
Squashed to 3

**Review by korydraughn — APPROVED** [review]
Please pound and merge when ready.

**Pull request merged**
